addons/website_sale_comparison/tests/test_website_sale_comparison.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import odoo.tests
+
+
+@odoo.tests.tagged('-at_install', 'post_install')
+class TestUi(odoo.tests.HttpCase):
+    def test_01_tour_website_sale_comparison_remove(self):
+        """ This tour makes sure the product page still works after the module
+        `website_sale_comparison` has been removed.
+
+        Technically it tests the removal of copied views by the base method
+        `_remove_copied_views`. The problematic view that has to be removed is
+        `product_add_to_compare` because it has a reference to `add_to_compare`.
+        """
+
+        # first test that the comparison button is present initially and
+        # actually create a copy of the view by using the editor to trigger COW
+        # on the product page and all its tree (including the view we want)
+        self.phantom_js("/", "odoo.__DEBUG__.services['web_tour.tour'].run('tour_shop_product_page_comparison')", "odoo.__DEBUG__.services['web_tour.tour'].tours.tour_shop_product_page_comparison.ready", login="admin")
```

Comment:
Can't all this be done in python?
```python
# Retrieve the generic view
generic_view = View.with_context(website_id=None).get_view_id('website_sale.product')
# Trigger COW
generic_view.with_context(website=1).write({'name': 'Trigger COW'})
```
Then remove the module

And next JS tour can simply be replaced by

```python
# Get specific view DOM
specific_view = View.with_context(website_id=1).get_view_id('website_sale.product')
specific_view_arch = View.browse(specific_view).with_context(load_all_views=True, website_id=1).read_combined(['arch'])['arch']
# Check compare is not in it anymore
self.assertEqual('compare' in specific_view_arch, False, "There should not be compare stuff left as we uninstalled the module")
```

Reply:
Well I think the question here is do we want to test an actual flow, or do we want to unit test the method?

The test I wrote is almost exactly matching what I did functionally to get the issue, which I think it is nice to cover.

But I would also be in favor of unit testing the method by adding a test more or less like you suggest in base.

Reply:
Honestly I feel like it would be better to test the uninstall module (triggering cow & checking after uninstall if compare is not there anymore) in python like I suggested.

And add a test to test comparison if needed but keep both tests separated.

Would be clearer that way, each test testing its own stuff. And in the future if we need to change the comparison tour (make it bigger, test new stuff etc) we wont have the view uninstall during removal at the same time

Reply:
Ok as we saw together, we adapted those snippets and we'll keep them instead.